mailhog-command.php

```diff
@@ -10,3 +10,12 @@
 }
 
 EE::add_command( 'mailhog', 'Mailhog_Command' );
+
+EE::add_hook('before_invoke:mailhog up', function ( $args, $assoc_args ) {
+
+	if ( ! is_array( EE::get_runner()->find_command_to_run( [ 'auth' ] ) ) ) {
+		EE::error( 'Auth command needs to be registered for mailhog' );
+	}
+
+	EE::run_command( [ 'auth', 'init' ] );
```

Comment:
here, please do not run it via command, when you can check for the class/function and directly execute it. 
Here our process might go slow in booting and run the command.

Reply:
@rahulsprajapati But what if we add hook on `auth init` in future? We would then have to call it manually. This would take care of it.

> Here our process might go slow in booting and run the command.

It could be hardly in milliseconds. I don't think it would even be noticeable.

Reply:
>It could be hardly in milliseconds. I don't think it would even be noticeable.
Well every major operation mostly in milliseconds only, it's up to us if that it is worth spending that operation or not.

Here when we run the command you can check all the process it's doing internally by `--debug` and after all that process which is anyway not going to be in use, you are calling that function only which does the job. Why do you want to spend such extra process here? 

---

> But what if we add a hook on auth init in future? We would then have to call it manually. This would take care of it.

We can take care of it here, executing code by adding multiple time ( from the common function ) is makes more sense then running entire command process which we do not require. Also, here our goal is just to execute code for `init`.

Reply:
> We can take care of it here, executing code by adding multiple time.

Yes that can be done by calling that method manually. But the trouble is that someone who is adding hook `admin init` in future may not know he has to call that function manually here. He would assume it would be called every time function is called. This *might* cause subtle bugs which may be hard to catch and hence it will affect maintainability of the project. 

I know this would cost minor performance penalty, but I think this is a trade off we should consider doing, so that maintainability of project isn't affected in long run.

Reply:
`do_hook( 'before_invoke:auth init' )` Seems like a good solution for now. Adding it.